app/factory.py

```diff
@@ -43,23 +43,27 @@ def get_user(self, identifier):
             if rv is not None:
                 return rv
 
+user_datastore = DeploySQLAlchemyUserDatastore(db, User, Role)
 
-def create_app(): #pylint: disable=too-many-statements
+def create_app(config=None): #pylint: disable=too-many-statements
     '''
     Create an instance of the app.
     '''
     app = Flask(__name__)
 
-    with open('/noi/app/config/config.yml', 'r') as config_file:
-        app.config.update(yaml.load(config_file))
+    if config is None:
+        with open('/noi/app/config/config.yml', 'r') as config_file:
+            app.config.update(yaml.load(config_file))
 
-    app.config['CELERYBEAT_SCHEDULE'] = CELERYBEAT_SCHEDULE
+        app.config['CELERYBEAT_SCHEDULE'] = CELERYBEAT_SCHEDULE
 
-    try:
-        with open('/noi/app/config/local_config.yml', 'r') as config_file:
-            app.config.update(yaml.load(config_file))
-    except IOError:
-        app.logger.warn("No local_config.yml file")
+        try:
+            with open('/noi/app/config/local_config.yml', 'r') as config_file:
+                app.config.update(yaml.load(config_file))
+        except IOError:
+            app.logger.warn("No local_config.yml file")
+    else:
+        app.config.update(config)
```

Comment:
Hmm, is this how we want to do it?  I can understand ignoring `local_config.yml`, but it seems that `config.yml` may have defaults we want to keep & override one-by-one through the dict passed to `create_app`.